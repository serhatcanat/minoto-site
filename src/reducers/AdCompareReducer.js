import {ADD_VEHICLE_TO_COMPARE, DELETE_VEHICLE_FROM_COMPARE} from "../actions/actionTypes";
import {CompareListService,LocalStorageService,LocalStorageItem} from '../functions'

const _localStorageService = new LocalStorageService();
const _compareListService = new CompareListService();
const INIT_STATE = {
    data: _localStorageService.get(LocalStorageItem.compareList) || [],
};


const addVehicleToCompareSuccess = (state, action) => {

    if(_compareListService.isFull(state.data)){
        state.data.pop();
        return ({
            ...state,
            data: [action.payload,...state.data]
        });
    }

    return ({
        ...state,
        data: [...state.data, action.payload]
    });
    // return state;
};

const deleteVehicleFromCompareSuccess = (state, action) => {
    const filteredItems = state.data.filter(item => item.id !== action.payload);
    return ({
        data: filteredItems
    });
};

const createReducer = (initialState, handlers) => {
    return (state = initialState, action) => {
        return handlers.hasOwnProperty(action.type)
            ? handlers[action.type](state, action)
            : state;
    }
};

export const adCompareReducer = createReducer(INIT_STATE, {
    [ADD_VEHICLE_TO_COMPARE]: addVehicleToCompareSuccess,
    [DELETE_VEHICLE_FROM_COMPARE]: deleteVehicleFromCompareSuccess,
});
