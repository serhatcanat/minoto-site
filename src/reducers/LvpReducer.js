import {ADD_VEHICLE_TO_LVP} from "../actions/actionTypes";
import {LVPService, LocalStorageService, LocalStorageItem, CompareListService} from '../functions'

const _localStorageService = new LocalStorageService();
const _compareListService = new CompareListService();

const INIT_STATE = {
    data: _localStorageService.get(LocalStorageItem.lvpList) || [],
};

const addVehicleToLVPSuccess = (state, action) => {
    if(_compareListService.isFull(state.data,6)){
        state.data.pop();
        return ({
            ...state,
            data: [action.payload,...state.data]
        });
    }

    return ({
        ...state,
        data: [...state.data, action.payload]
    });
    // return state;
};

const createReducer = (initialState, handlers) => {
    return (state = initialState, action) => {
        return handlers.hasOwnProperty(action.type)
            ? handlers[action.type](state, action)
            : state;
    }
};


export const lvpReducer = createReducer(INIT_STATE, {
    [ADD_VEHICLE_TO_LVP]: addVehicleToLVPSuccess,
});
