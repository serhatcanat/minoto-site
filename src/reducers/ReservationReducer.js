import {SET_VEHICLE_TO_RESERVATION} from "../actions/actionTypes";

const INIT_STATE = {
    data: [],
};

const setVehicleToReservation = (state, action) => {
    return ({
        data: [action.payload]
    });
};

const createReducer = (initialState, handlers) => {
    return (state = initialState, action) => {
        return handlers.hasOwnProperty(action.type)
            ? handlers[action.type](state, action)
            : state;
    }
};

export const reservationReducer = createReducer(INIT_STATE, {
    [SET_VEHICLE_TO_RESERVATION]: setVehicleToReservation
});
