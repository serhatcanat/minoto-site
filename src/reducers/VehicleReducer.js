import {GET_VEHICLE_DETAIL_SUCCESS} from "../actions/actionTypes";

const INIT_STATE = {
    data: [],
};

const getVehicleDetail = (state, action) => ({
    ...state,
    data: action.payload
});


const createReducer = (initialState, handlers) => {
    return (state = initialState, action) => {
        return handlers.hasOwnProperty(action.type)
            ? handlers[action.type](state, action)
            : state;
    }
};

export const vehicleReducer = createReducer(INIT_STATE, {
    [GET_VEHICLE_DETAIL_SUCCESS]: getVehicleDetail
});
