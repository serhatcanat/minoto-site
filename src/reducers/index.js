import {combineReducers} from 'redux';
import {vehicleReducer} from "./VehicleReducer";
import {adCompareReducer} from "./AdCompareReducer";
import {reservationReducer} from "./ReservationReducer";


export default () => combineReducers({
    adCompare: adCompareReducer,
    vehicle: vehicleReducer,
    reservation: reservationReducer,
});
