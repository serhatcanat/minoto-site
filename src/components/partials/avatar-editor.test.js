import MyAvatar from './avatar-editor'
import React from 'react'
import {adCompareReducer} from "../../reducers/AdCompareReducer";
import Adapter from 'enzyme-adapter-react-16';
import {shallow,configure} from "enzyme";

configure({ adapter: new Adapter() });

describe('<MyAvatar />', () => {
    it('renders an editor area', () => {
        const editor = shallow(<MyAvatar/>);
        expect(editor.find('.form-inputwrap'))
    })
})
