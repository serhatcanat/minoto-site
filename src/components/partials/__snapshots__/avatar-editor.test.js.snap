// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyAvatar /> renders an editor area 1`] = `
"<div className=\\"inputwrap type-image form-inputwrap\\">
  <label className=\\"input-uploadlabel\\">
    Yeni dosya yüklemek için tıklayın
    <input type=\\"file\\" className=\\"inputFile\\" onChange={[Function]} value=\\"\\" />
  </label>
</div>"
`;
